backend-lib/build.rs

```diff
@@ -0,0 +1,55 @@
+use std::env;
+use std::path::PathBuf;
+
+const DEFAULT_CLANG_VERSION: &str = "17";
+
+fn main() {
+    setup_x86_64_android_workaround();
+}
+
+/// Adds a temporary workaround for an issue with the Rust compiler and Android in x86_64
+/// devices: https://github.com/rust-lang/rust/issues/109717.
+/// The workaround comes from: https://github.com/mozilla/application-services/pull/5442
+fn setup_x86_64_android_workaround() {
```

Comment:
It would be useful to document the ultimate approach at a high level here, as figuring it out from the links provided requires a bunch of digging (and as we know, links aren't forever.)

Reply:
Done.